### PR TITLE
Simplify CD steps

### PR DESCRIPTION
**Pull request by BPerlakiH**
Trying to simplify the CD steps to be more readable before updating it to support the requests in the ticket.

### CODE REVIEW

File: .github/workflows/cd.yml [hunk 1]
```diff
@@ -61,61 +63,41 @@ jobs:
...
-        APPLE_DEVELOPER_ID_SIGNING_P12_PASSWORD: ${{ secrets.APPLE_DEVELOPER_ID_SIGNING_P12_PASSWORD }}
-        APPLE_DEVELOPER_ID_SIGNING_IDENTITY: ${{ secrets.APPLE_DEVELOPER_ID_SIGNING_IDENTITY }}
-      run: |
-        echo "SIGNING_CERTIFICATE=${APPLE_DEVELOPER_ID_SIGNING_CERTIFICATE}" >> "$GITHUB_ENV"
-        echo "SIGNING_CERTIFICATE_P12_PASSWORD=${APPLE_DEVELOPER_ID_SIGNING_P12_PASSWORD}" >> "$GITHUB_ENV"
-        echo "SIGNING_IDENTITY=${APPLE_DEVELOPER_ID_SIGNING_IDENTITY}" >> "$GITHUB_ENV"
+        SIGNING_CERTIFICATE: ${{ secrets.APPLE_DEVELOPER_ID_SIGNING_CERTIFICATE }}
+        SIGNING_CERTIFICATE_P12_PASSWORD: ${{ secrets.APPLE_DEVELOPER_ID_SIGNING_P12_PASSWORD }}
+        SIGNING_IDENTITY: ${{ secrets.APPLE_DEVELOPER_ID_SIGNING_IDENTITY }}
+      run: echo ""
```
Comment (rgaudin) [inline .github/workflows/cd.yml:79]: This does nothing. What's the point? Those `env` are per-step and not persisted

File: .github/workflows/cd.yml [hunk 2]
```diff
@@ -61,61 +63,41 @@ jobs:
...
           fh.write(f"ISRELEASE={'yes' if is_release else ''}\n")
           fh.write(f"EXPORT_METHOD={export_method}\n")
           fh.write(f"UPLOAD_FOLDER={upload_folder}\n")
-          fh.write(f"EXTRA_XCODEBUILD={extra_xcode}\n")
           fh.write(f"UPLOAD_TO_APPLE={'yes' if upload_to_apple else ''}\n")
 
     - name: Prepare use of Developper ID Certificate
-      if: ${{ matrix.destination.uploadto == 'dmg' }}
-      shell: bash
+      if: matrix.destination.uploadto == 'dmg'
```
Comment (rgaudin) [inline .github/workflows/cd.yml:74]: Please harmonize with CI. You changed `dmg` to `ftp` there for some reason so use same naming in both workflows

**Review by rgaudin — CHANGES_REQUESTED** [review]

File: .github/workflows/cd.yml (hunk 1 above)
Comment (BPerlakiH) [reply .github/workflows/cd.yml]: Thanks, I have updated it.

**Comment by kelson42** [comment]
@BPerlakiH Is that really still a draft?

**Comment by BPerlakiH** [comment]
> @BPerlakiH Is that really still a draft?

I just double checked the CD build, and it is currently failing with the changes on this branch, so I marked it as a draft, not to merge it accidentally.
Once I fix it, and both CI and CD jobs run OK, will change it back.

File: .github/workflows/cd.yml (hunk 2 above)
Comment (BPerlakiH) [reply .github/workflows/cd.yml]: Please see it updated.

**Comment by BPerlakiH** [comment]
It is ready for review.
The FTP upload part works fine:
Screenshot 2024-02-07 at 13 33 57

File: .github/workflows/cd.yml [hunk 3]
```diff
@@ -151,64 +125,60 @@ jobs:
...
-      if: ${{ matrix.destination.uploadto == 'ipa' }}
+      if: matrix.platform == 'iOS' && env.UPLOAD_TO == 'ftp'
       run: |
         plutil -create xml1 ./export.plist
         plutil -insert method -string ad-hoc ./export.plist
         plutil -insert provisioningProfiles -dictionary ./export.plist
         plutil -replace provisioningProfiles -json '{ "self.Kiwix" : "iOS Team Provisioning Profile" }' ./export.plist
 
+    - name: Prepare export for ${{ matrix.uploadto }}
+      if: matrix.platform != 'iOS' || env.UPLOAD_TO == 'app-store'
```
Comment (rgaudin) [inline .github/workflows/cd.yml:137]: condition is hard to read ; can you add a comment describing the use cases?

File: .github/workflows/cd.yml [hunk 4]
```diff
@@ -151,64 +125,60 @@ jobs:
...
+
     - name: Upload Archive to Apple (App Store or Notarization)
-      if: ${{ env.UPLOAD_TO_APPLE }}
+      if: env.UPLOAD_TO != ''
       env:
-        APPLE_STORE_AUTH_KEY_PATH: ${{ env.APPLE_STORE_AUTH_KEY_PATH }}
-        APPLE_STORE_AUTH_KEY_ID: ${{ secrets.APPLE_STORE_AUTH_KEY_ID }}
-        APPLE_STORE_AUTH_KEY_ISSUER_ID: ${{ secrets.APPLE_STORE_AUTH_KEY_ISSUER_ID }}
-      run: python .github/retry-if-retcode.py --sleep 60 --attempts 5 --retcode 70 xcrun xcodebuild -exportArchive -archivePath $PWD/Kiwix-$VERSION.xcarchive -exportPath $PWD/export/ -exportOptionsPlist export.plist -authenticationKeyPath $APPLE_STORE_AUTH_KEY_PATH -allowProvisioningUpdates -authenticationKeyID $APPLE_STORE_AUTH_KEY_ID -authenticationKeyIssuerID $APPLE_STORE_AUTH_KEY_ISSUER_ID
+        VERSION: ${{ env.VERSION }}
```
Comment (rgaudin) [inline .github/workflows/cd.yml:146]: This is useless and confusing

File: .github/workflows/cd.yml [hunk 5]
```diff
@@ -151,64 +125,60 @@ jobs:
...
+      run: python .github/retry-if-retcode.py --sleep 60 --attempts 20 --retcode 65 xcrun xcodebuild -exportNotarizedApp -archivePath $PWD/Kiwix-$VERSION.xcarchive -exportPath $PWD/export/ -allowProvisioningUpdates ${{ env.APPLE_AUTH_PARAMS }}
 
-    - name: Create DMG
-      if: ${{ matrix.destination.uploadto == 'dmg' }}
-      run: |
+    - name: Create and Notarize DMG
+      if: matrix.platform == 'macOS' && env.UPLOAD_TO == 'ftp'
+      env:
+        VERSION: ${{ env.VERSION }}
```
Comment (rgaudin) [inline .github/workflows/cd.yml:158]: idem

File: .github/workflows/cd.yml [hunk 6]
```diff
@@ -151,64 +125,60 @@ jobs:
...
-        echo "${PRIVATE_KEY}" > $SSH_KEY
+        echo "${{ secrets.SSH_KEY }}" > $SSH_KEY
         chmod 600 $SSH_KEY
 
     - name: Upload DMG
-      if: ${{ matrix.destination.uploadto == 'dmg' }}
-      run: python .github/upload_file.py --src ${PWD}/kiwix-${VERSION}.dmg --dest ci@master.download.kiwix.org:30022/data/download/${UPLOAD_FOLDER} --ssh-key ${SSH_KEY}
+      if: env.UPLOAD_TO == 'ftp' && matrix.platform == 'macOS'
+      env:
+        VERSION: ${{ env.VERSION }}
```
Comment (rgaudin) [inline .github/workflows/cd.yml:174]: And again

File: .github/workflows/cd.yml [hunk 7]
```diff
@@ -151,64 +125,60 @@ jobs:
...
+      run: |
+        python .github/upload_file.py --src ${PWD}/kiwix-${VERSION}.dmg --dest ci@master.download.kiwix.org:30022/data/download/${UPLOAD_FOLDER} --ssh-key ${SSH_KEY}
 
     - name: Upload IPA
-      if: ${{ matrix.destination.uploadto == 'ipa' }}
+      if: env.UPLOAD_TO == 'ftp' && matrix.platform == 'iOS'
+      env:
+        VERSION: ${{ env.VERSION }}
```
Comment (rgaudin) [inline .github/workflows/cd.yml:181]: (!)

File: .github/workflows/ci.yml [hunk 8]
```diff
@@ -20,54 +17,38 @@ jobs:
...
-          import os
-          extra_xcode = os.getenv("EXTRA_XCODEBUILD", "")
-          extra_xcode += f" -authenticationKeyPath {os.getenv('APPLE_STORE_AUTH_KEY_PATH')}"
-          extra_xcode += f" -authenticationKeyID {os.getenv('APPLE_STORE_AUTH_KEY_ID')}"
-          extra_xcode += f" -authenticationKeyIssuerID {os.getenv('APPLE_STORE_AUTH_KEY_ISSUER_ID')}"
-
-          with open(os.getenv("GITHUB_ENV"), "a") as fh:
-            fh.write(f"EXTRA_XCODEBUILD={extra_xcode}\n")
+          EXTRA_XCODEBUILD: "-sdk iphoneos ${{ env.APPLE_AUTH_PARAMS }}"
+        run: echo
```
Comment (rgaudin) [inline .github/workflows/ci.yml:44]: What's this? Does nothing. Wont set that env for later steps

**Review by rgaudin — CHANGES_REQUESTED** [review]
Thank you ; please see inline comments

File: .github/workflows/cd.yml (hunk 3 above)
Comment (BPerlakiH) [reply .github/workflows/cd.yml:140]: Please see it updated.

File: .github/workflows/cd.yml (hunk 4 above)
Comment (BPerlakiH) [reply .github/workflows/cd.yml:149]: Removed it.

File: .github/workflows/cd.yml (hunk 6 above)
Comment (BPerlakiH) [reply .github/workflows/cd.yml]: Removed it.

File: .github/workflows/cd.yml (hunk 7 above)
Comment (BPerlakiH) [reply .github/workflows/cd.yml]: Removed it.

**Review by rgaudin — APPROVED** [review]
LGTM now

**Pull request merged**